chore(eslint-config): update 'typescript-eslint' to v8.43.0 and replace deprecated 'tseslint.config' with 'defineConfig'

sukvvon opened the pull request:
# Overview

<!--
    A clear and concise description of what this pr is about.
 -->

https://typescript-eslint.io/packages/typescript-eslint/#migrating-to-defineconfig

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

sukvvon commented on pnpm-lock.yaml:
I resolved the TypeScript error in the `defineConfig` of `vite.config.ts` in the `vite-react-18-suspense-prerender-siblings-problem` project, which was caused by dependency issues, through a clean install to fix the dependency problems.

manudeli commented:
Could you resolve the conflict, please?

manudeli commented on packages/react/CHANGELOG.md:
```diff
@@ -21,7 +21,6 @@
   This adds OIDC (OpenID Connect) authentication support for npm publishing in GitHub Actions. The implementation enables secure, tokenless publishing to npm using GitHub's OIDC tokens with provenance support.
 
   Key features:
-
```
remove unnecessary change please

manudeli commented on packages/react-query/src/v5.ts:
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-deprecated */
```
```suggestion
```

Could you make @typescript-eslint/no-deprecated as warn in @suspensive/eslint-config?

manudeli commented on packages/react-query/src/index.ts:
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-deprecated */
```
```suggestion
```

manudeli commented on packages/react-query-5/src/index.ts:
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-deprecated */
```
```suggestion
```

manudeli commented on packages/codemods/src/transforms/testfixtures/migrate-query-client-consumer-props/migrate-query-client-consumer-props.output.jsx:
```diff
@@ -6,7 +6,7 @@ const queryClientContext = createContext(queryClient)
 
 const PostRefreshButton = () => {
   return (
-    (<QueryClientConsumer queryClient={queryClient}>
```
👍

manudeli commented on packages/react-query-4/CHANGELOG.md:
```diff
@@ -17,7 +17,6 @@
   This adds OIDC (OpenID Connect) authentication support for npm publishing in GitHub Actions. The implementation enables secure, tokenless publishing to npm using GitHub's OIDC tokens with provenance support.
 
   Key features:
-
```
Remove unnecessary change please

manudeli replied:
Could you remove unnecessary change? this wouldn't be autofixed

manudeli commented on packages/react-native/CHANGELOG.md:
```diff
@@ -15,7 +15,6 @@
   This adds OIDC (OpenID Connect) authentication support for npm publishing in GitHub Actions. The implementation enables secure, tokenless publishing to npm using GitHub's OIDC tokens with provenance support.
 
   Key features:
-
```
Remove unnecessary change please

sukvvon commented on packages/react-query/CHANGELOG.md:
```diff
@@ -27,7 +27,6 @@
   This adds OIDC (OpenID Connect) authentication support for npm publishing in GitHub Actions. The implementation enables secure, tokenless publishing to npm using GitHub's OIDC tokens with provenance support.
 
   Key features:
-
```
<img width="957" height="586" alt="image" src="https://github.com/user-attachments/assets/927dbda6-5640-49a2-8a2e-dc6ae2188eec" />

I can't find `-`

sukvvon commented:
@manudeli I think autofix.ci do this

sukvvon commented:
> > [f2f5b69](https://github.com/toss/suspensive/commit/f2f5b69c28b633be14fc9df7cc301ff1a4a7146b)
> 
> @sukvvon I updated this branch (I added `packages/**/CHANGELOG.md` in .prettierignore) so could you remove unnecessary changes please?

@manudeli I've done

manudeli reviewed (approved):
Thanks!
